### PR TITLE
Octavia support for JS 13.1

### PR DESCRIPTION
**Pull request by rashidxf**
Added a flag in settings.ini files to enable/disable Octavia load balancer at deployment time.
Also, provided an option for Users to disable Octavia generated certificates and keys for secure communication between Octavia containers and amphora load balancer and provide their own certificates and keys in settings.ini files.

### CODE REVIEW

File: src/deploy/osp_deployer/director.py [hunk 1]
```diff
@@ -678,10 +678,25 @@ def setup_net_envt(self):
...
         self.upload_file(self.settings.network_env_yaml,
                          network_yaml)
-
+                         
+        if self.settings.octavia_user_certs_keys is True:
+          self.upload_file(self.settings.certificate_keys_path,
+                           self.templates_dir + "/cert_keys.yaml")
+          self.run_tty('sed -i "s|OctaviaGenerateCerts:.*|' + 
+            'OctaviaGenerateCerts: ' +
+            'False' + '|" ' +
```
Comment (gaelrehault) [inline src/deploy/osp_deployer/director.py]: can you make is lowercase (false) to match up with default template s case

File: src/deploy/osp_deployer/director.py [hunk 2]
```diff
@@ -678,10 +678,25 @@ def setup_net_envt(self):
...
+                           self.templates_dir + "/cert_keys.yaml")
+          self.run_tty('sed -i "s|OctaviaGenerateCerts:.*|' + 
+            'OctaviaGenerateCerts: ' +
+            'False' + '|" ' + 
+            str(octavia_yaml))
+        
+        if self.settings.octavia_user_certs_keys is False:
+          self.run_tty('sed -i "s|OctaviaGenerateCerts:.*|' + 
+            'OctaviaGenerateCerts: ' +
+            'True' + '|" ' +
```
Comment (gaelrehault) [inline src/deploy/osp_deployer/director.py]: can you make is lowercase (true) to match up with default template s case

File: src/deploy/osp_deployer/director.py (hunk 1 above)
Comment (rashidxf) [reply src/deploy/osp_deployer/director.py]: this has been fixed

File: src/deploy/osp_deployer/director.py (hunk 2 above)
Comment (rashidxf) [reply src/deploy/osp_deployer/director.py]: this has been fixed

File: src/pilot/deploy-overcloud.py [hunk 3]
```diff
@@ -464,6 +472,14 @@ def main():
...
             env_opts += " -e ~/pilot/templates/neutron-ovs-dvr.yaml"
 
+        # The octavia.yaml must be included after the
+        # network-environment.yaml
+        if args.octavia_enable:
+            env_opts += " -e ~/pilot/templates/octavia.yaml"          
+
+        if args.octavia_user_certs_keys is True:
+            env_opts += " -e ~/pilot/templates/cert_keys.yaml"
+
```
Comment (gaelrehault) [inline src/pilot/deploy-overcloud.py]: You probably only want to enable the cert yaml if octavia is enabled as well.

Comment (rashidxf) [reply src/pilot/deploy-overcloud.py]: Yes you are right. This is fixed now.

**Pull request merged**